canopeum_frontend/src/components/social/site-modal/SiteAnnouncementModal.tsx

```diff
@@ -0,0 +1,110 @@
+import { Dialog, DialogActions, DialogContent, DialogTitle } from '@mui/material'
+import { useContext, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { SnackbarContext } from '@components/context/SnackbarContext'
+import UrlTextField from '@components/inputs/UrlTextField'
+import useApiClient from '@hooks/ApiClientHook'
+import type { Announcement, PatchedAnnouncement } from '@services/api'
+
+type Props = {
+  readonly announcement: Announcement,
+  readonly isOpen: boolean,
+  readonly handleClose: (contact: Announcement | null) => void,
+}
+
+type EditSiteAnnouncementDto = {
+  body?: string,
+  link?: string,
+}
+
+const SiteAnnouncementModal = ({ announcement, isOpen, handleClose }: Props) => {
+  const { t } = useTranslation()
+  const [editedAnnouncement, setEditedAnnouncement] = useState<EditSiteAnnouncementDto>(
+    announcement,
+  )
+  const [isFormValid, setIsFormValid] = useState<boolean>(true)
+  const { getApiClient } = useApiClient()
+  const { openAlertSnackbar } = useContext(SnackbarContext)
+
+  const handleSubmitSiteAnnouncement = async () => {
+    try {
+      await getApiClient().announcementClient.update(
+        announcement.id,
+        editedAnnouncement as PatchedAnnouncement,
+      )
+    } catch (error: unknown) {
+      console.error(error)
```

Comment:
This is a warning because the linter is telling you: "Maybe this error should be shown to the user instead of hidden in logs?". But we already use the snackbar. Maybe using a "console.error" for snackbar errors in general could be a good idea.